src/react-components/ory/helpers/node.tsx

```diff
@@ -56,47 +56,131 @@ export const getNodeLabel = (node: UiNode): UiText | undefined => {
+ *      "type":"info",
+ *      "context":\{
+ *        "title":"E-Mail"
+ *      \},
+ *    \}
+ *  \}
+ * \}
+ *
+ * The context has the key "title" which matches the formatter template name "\{title\}"
+ * An example translation file would look like this:
+ * \{
+ *  "identities.messages.1070002": "\{title\}"
+ * \}
+ *
+ * The formwatter would then take the meta.label.id and look for the translation with the key matching the id.
+ * It would then replace the template "\{title\}" with the value from the context with the key "title".
+ *
+ * @param uiText - The UiText is part of the UiNode object sent by Kratos when performing a flow.
+ */
 export const uiTextToFormattedMessage = (
   { id, context = {}, text }: Omit<UiText, "type">,
   intl: IntlShape,
-) =>
-  intl.formatMessage(
+) => {
+  const contextInjectedMessage = Object.entries(context).reduce(
+    (accumulator, [key, value]) => {
+      // context might provide an array of objects instead of a single object
+      // for example when looking up a recovery code
+      /*
+      *
+      {
+      "text": {
+          "id": 1050015,
+          "text": "3r9noma8, tv14n5tu, ...",
+          "type": "info",
+          "context": {
+              "secrets": [
+                  {
+                      "context": {
+                          "secret": "3r9noma8"
+                      },
+                      "id": 1050009,
+                      "text": "3r9noma8",
+                      "type": "info"
+                  },
+                  {
+                      "context": {
+                          "secret": "tv14n5tu"
+                      },
+                      "id": 1050009,
+                      "text": "tv14n5tu",
+                      "type": "info"
+                  },
+              ]
+          }
+      },
+      "id": "lookup_secret_codes",
+      "node_type": "text"
+      }
+      */
+      if (Array.isArray(value)) {
+        return {
+          ...accumulator,
+          [key]: value,
+          [key + "_list"]: intl.formatList<string>(value),
+        }
+      } else if (key.endsWith("_unix")) {
+        if (typeof value === "number") {
+          return {
+            ...accumulator,
+            [key]: intl.formatDate(new Date(value * 1000)),
+            [key + "_since"]: intl.formatDateTimeRange(
+              new Date(value),
+              new Date(),
+            ),
+            [key + "_since_minutes"]: Math.abs(
+              (value - new Date().getTime() / 1000) / 60,
+            ).toFixed(2),
+            [key + "_until"]: intl.formatDateTimeRange(
+              new Date(),
+              new Date(value),
+            ),
+            [key + "_until_minutes"]: Math.abs(
+              (new Date().getTime() / 1000 - value) / 60,
+            ).toFixed(2),
+          }
+        }
+      }
+      return {
+        ...accumulator,
+        [key]: value as string | number,
+      }
+    },
+    {},
+  )
+
+  return intl.formatMessage(
```

Comment:
This needs some review. I am changing the date format as well here to only have 2 decimal places in the minutes we display. For example "the flow has expired 10.00 minutes ago".

Right now it's a bit unwieldy and would display "the flow has expired -10.0004123 minutes ago".

Reply:
There are some tests inside the error-messages.spec file and inside the node.spec.tsx file.

Reply:
Probably makes more sense to fix this in kratos?

Reply:
kratos gives back a unix timestamp from where it expired. So the date. 
The ui then has to render the time or the minutes from/until that date.

Reply:
Ah I get it, this basically formats the timestamp in separate keys (until, since, ...). Makes sense to me

Reply:
yeah exactly. Do you think it's fine to merge?